### DISCUSSION TITLE
how to make user interact with ACAP app (built by Native SDK)?

### DISCUSSION THREAD

**Discussion by eliangbb** [AxisCommunications/discussions#32] [ACAP] [answered]
After installing or starting/running an app under camera/device web page, we are trying to ask user to input something. Just like Axis built-in app "Axis Loitering Guard", user could click "Open" button and use Profile to setup some parameters. Or something like ACAP native SDK sample app "licensekey", after installation, user could click top right 3-dot button to choose "Activate license with a key" to allow user to enter key. We want to do something similarly, allowing user to interact with app. Is there a way to do so? Do you have a sample application code for this? Thx.

**Comment by pataxis** [comment, marked as answer]
The old solution which **Axis Loitering Guard** most probably is using is the `axparameter` API, but it has been removed in ACAP Native SDK (ACAP 4). For more information of the future of the `ax*` APIs like `axparameter`, see https://axiscommunications.github.io/acap-documentation/docs/introduction/acap4-vs-acap3.html .

License key handling is a built-in feature in Axis OS, not the application. In other words the `licensekey` example is not using `axparameter`.

One solution could be to use the web-server example, see the [C API examples](https://github.com/AxisCommunications/acap-native-sdk-examples/tree/main/web-server#c-api-examples) section for some examples of what can be done. If your application handles it's parameters itself, you should be able to expose them through the web-server solution. For storing parameters e.g. to survive a reboot, the [localdata](https://axiscommunications.github.io/acap-documentation/docs/develop/app-project-structure.html#local-data) directory could be used.

**Reply by eliangbb** [reply]
will try. thx.
